lib/ansible/modules/windows/win_file.ps1

```diff
@@ -20,51 +20,601 @@
+    }
+}
 
 $result = @{
     changed = $false
 }
 
-# Used to delete symlinks as powershell cannot delete broken symlinks
+# Used to interact with Links and windows
+# Powershell does have native link handling through New-Item but was
+# only added with V5 and Server 2016/Windows 10 onwards
 $symlink_util = @"
 using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Win32.SafeHandles;
+using System.IO;
 using System.ComponentModel;
 using System.Runtime.InteropServices;
 
-namespace Ansible.Command {
-    public class SymLinkHelper {
-        [DllImport("kernel32.dll", CharSet=CharSet.Unicode, SetLastError=true)]
+namespace Ansible
+{
+    public class SymLink
+    {
+        /*
+         * Parts of this code has been derived from various sources
+         * 
+         * Jeff Brown - Creating Junction Points https://www.codeproject.com/Articles/15633/Manipulating-NTFS-Junction-Points-in-NET
+         * Dave Midgley - Reparse Points in Vista https://www.codeproject.com/kb/vista/reparsepointid.aspx
+         * 
+         * Thanks to the above for help with structuring the processes
+         * that is required
+         * 
+         */
+
+        // Types of Links
+        private const string SYMBOLIC_LINK = "link";
+        private const string HARD_LINK = "hard";
+        private const string JUNCTION_POINT = "junction";
+        private const string MOUNT_POINT = "mount_point";
+
+        // Constants used when setting SeBackupPrivilege
+        private const uint TOKEN_ADJUST_PRIVILEGES = 0x00000020;
+        private const uint SE_PRIVILEGE_ENABLED = 0x00000002;
+        private const string SE_BACKUP_NAME = "SeBackupPrivilege";
+
+        // Reparse Tags https://msdn.microsoft.com/en-US/library/windows/desktop/aa365511(v=vs.85).aspx - Values derived from WinNT.h
+        private const uint IO_REPARSE_TAG_MOUNT_POINT = 0xA0000003;
+        private const uint IO_REPARSE_TAG_SYMLINK = 0xA000000C;
+
+        // Flag set when the Soft Link path is a relative path
+        private const uint SYMLINK_FLAG_RELATIVE = 0x00000001;
+
+        // CTL_CODE Values
+        private const int FSCTL_GET_REPARSE_POINT = 0x000900A8;
+        private const int FSCTL_SET_REPARSE_POINT = 0x000900A4;
+        private const uint FILE_DEVICE_FILE_SYSTEM = 0x00090000;
+        private const uint FILE_FLAG_BACKUP_SEMANTICS = 0x02000000;
+        private const uint FILE_FLAG_OPEN_REPARSE_POINT = 0x00200000;
+        private const uint FILE_ATTRIBUTE_ARCHIVE = 0x20000000;
+
+        // Used when getting the Reparse Data
+        private const int MAXIMUM_REPARSE_DATA_BUFFER_SIZE = 1024 * 16;
+
+        // Size of the REPARSE_DATA_BUFFER headers (ReparseTag (4) + ReparseDataLength (2) + Reserved (2))
+        private const int REPARSE_DATA_BUFFER_HEADER_SIZE = 8;
+
+        // Allows greater than 32767 characters in path when creating a Junction Point
+        private const string MAXIMUM_PATH_FIX_PREFIX = @"\??\";
```

Comment:
This is a poor choice for variable naming, `\??\` is used as a prefix when creating a ReparsePoint. There isn't really much documentation I could find from Microsoft but here is some info about it https://stackoverflow.com/questions/14329653/does-the-substitutename-string-in-the-pathbuffer-of-a-reparse-data-buffer-stru.

I'll move this down to the actual method used as it is only used once and `MAXIMUM_PATH_FIX_PREFIX` isn't really what it is used for.